public/app/features/explore/state/query.ts

```diff
@@ -486,11 +485,7 @@ async function handleHistory(
   queries: DataQuery[],
   exploreId: string
 ) {
-  const datasourceId = datasource.meta.id;
-  const nextHistory = updateHistory(history, datasourceId, queries);
-  dispatch(historyUpdatedAction({ exploreId, history: nextHistory }));
-
-  dispatch(addHistoryItem(datasource.uid, datasource.name, queries));
+  addHistoryItem(datasource.uid, datasource.name, queries);
```

Comment:
🤔 addHistoryItem is still a thunk so should be dispatched, right?

```suggestion
  dispatch(addHistoryItem(datasource.uid, datasource.name, queries));
```